operations/node_management/Rebuild_NCNs/Prepare_Master_Nodes.md

```diff
@@ -0,0 +1,146 @@
+
+### Step 2 - Determine if the master node being rebuilt is the first master node
+
+***IMPORTANT:*** The first master node is the node others contact to join the Kubernetes cluster. If this is the node being rebuilt, promote another master node to the initial node before proceeding.
+
+1. Fetch the defined first-master-hostname
+
+    ```bash
+    ncn-m# craysys metadata get first-master-hostname
+    ncn-m002
+    ```
+  
+    * If the node returned is not the one being rebuilt, proceed to the step which [stops etcd](#stop-the-etcd-service-on-the-master-node-being-removed) and skip the substeps here.
+
+1. Reconfigure the Boot Script Service \(BSS\) to point to a new first master node.
+
+    On any master or worker node:
+
+    ```bash
+   cray bss bootparameters list --name Global --format=json | jq '.[]' > Global.json
+   ```
+
+1. Edit the Global.json file and edit the indicated line.
+
+    Change the `first-master-hostname` value to another node that will be promoted to the first master node. For example, if the first node is changing from `ncn-m002` to `ncn-m001`, the line would be changed to the following:
+
+    ```text
+   "first-master-hostname": "ncn-m001",
+   ```
+
+1. Get a token to interact with BSS via the REST API.
+
+    ```bash
+   curl -i -s -k -H "Content-Type: application/json"  -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/apis/bss/boot/v1/bootparameters" -X PUT -d @./Global.json
+   ```
+
+    Ensure a good response, such as `HTTP CODE 200`, is returned in the `curl` output.
+
+1. Configure the newly promoted first master node so it is able to have other nodes join the cluster.
+
+    Use `ssh` to login to the newly-promoted master node chosen in the previous steps \(`ncn-m001` in this example\), copy/paste the following script to a file, and then execute it.
+
+    ```bash
+    #!/bin/bash
+     source /srv/cray/scripts/metal/lib.sh
+    export KUBERNETES_VERSION="v$(cat /etc/cray/kubernetes/version)"
+    echo $(kubeadm init phase upload-certs --upload-certs 2>&1 | tail -1) > /etc/cray/kubernetes/certificate-key
+    export CERTIFICATE_KEY=$(cat /etc/cray/kubernetes/certificate-key)
+    export MAX_PODS_PER_NODE=$(craysys metadata get kubernetes-max-pods-per-node)
+    export PODS_CIDR=$(craysys metadata get kubernetes-pods-cidr)
+    export SERVICES_CIDR=$(craysys metadata get kubernetes-services-cidr)
+    envsubst < /srv/cray/resources/common/kubeadm.yaml > /etc/cray/kubernetes/kubeadm.yaml
+     kubeadm token create --print-join-command > /etc/cray/kubernetes/join-command 2>/dev/null
```

Comment:
this line is extra-indented. This applies to several of the following lines in this script.